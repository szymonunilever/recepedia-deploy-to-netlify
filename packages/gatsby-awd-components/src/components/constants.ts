export enum KeyCode {
  Enter = 13,
  Space = 32,
  Tab = 9,
  Delete = 46,
  ArrowDown = 40,
  ArrowLeft = 37,
  ArrowRight = 39,
  ArrowUp = 38,
  Escape = 27,
}

export const IMAGE_SIZES = {
  QUIZ_OPTIONS: {
    QUIZ_SMALL: '(max-width: 768px) 50vw, (max-width: 1366px) 30vw, 400px',
    QUIZ_BIG: '(max-width: 1366px) 50vw, 800px',
  },
  RECIPE_LISTINGS: {
    STANDARD: '(max-width: 768px) 50vw, (max-width: 1366px) 30vw, 400px',
    NON_STANDARD: '(max-width: 768px) 100vw, (max-width: 1366px) 50vw, 500px',
    MEAL_PLANNER: '(max-width: 768px) 50vw, (max-width: 1366px) 30vw, 400px',
  },
  PAGE_LISTINGS: {
    CAROUSEL: '(max-width: 768px) 30vw, (max-width: 1366px) 25vw, 300px',
    TILES: '(max-width: 1366px) 50vw, 350px',
  },
};

