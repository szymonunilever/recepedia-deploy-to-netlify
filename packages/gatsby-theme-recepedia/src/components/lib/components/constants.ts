export enum KeyCode {
  Enter = 13,
  Space = 32,
  Tab = 9,
  Delete = 46,
  ArrowDown = 40,
  ArrowLeft = 37,
  ArrowRight = 39,
  ArrowUp = 38,
  Escape = 27,
}
