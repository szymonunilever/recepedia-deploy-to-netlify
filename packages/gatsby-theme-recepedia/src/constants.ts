/**
 * RecipePersonalizationFormula
 * template - query string which using for Recipe Personalization result.
 * where:
 * Q - intro Quiz Answers
 * MP - meal Planer Answers
 * #(digit) - number of question
 * ^(digit) - weight of condition
 * searchAttributes - fields identifications
 * */
export const RecipePersonalizationFormula: RecipePersonalizationFormulaProps = {
  template: ['(Q#2^3) AND (Q#4^2 OR Q#1)', '(Q#2^3)'],
  searchAttributes: {
    tags: 'tagGroups.tags.id',
    cookTime: 'recipeDetails.cookTime',
    serves: 'recipeDetails.serves',
  },
};

export const MealPlannerPersonalizationFormulas = {
  'pt-br': {
    template: [
      'MP#2 AND MP#1 AND MP#3 AND MP#5 AND MP#4',
      'MP#2 AND MP#1 AND MP#3 AND MP#5',
      'MP#2 AND MP#1 AND MP#3',
      'MP#2 AND MP#1',
    ],
    searchAttributes: RecipePersonalizationFormula.searchAttributes,
  },
  'es-mx': {
    template: [
      'MP#1 AND MP#3 AND MP#4 AND MP#2',
      'MP#1 AND MP#3 AND MP#4',
      'MP#1 AND MP#3',
      'MP#1',
    ],
    searchAttributes: RecipePersonalizationFormula.searchAttributes,
  },
};

export interface RecipePersonalizationFormulaProps {
  template: string[];
  searchAttributes: {
    [key: string]: string;
  };
}

export const IMAGE_SIZES = {
  QUIZ_OPTIONS: {
    QUIZ_4_IN_LINE: '(max-width: 768px) 40vw, (max-width: 1365px) 30vw, 300px',
    QUIZ_3_IN_LINE: '(max-width: 768px) 40vw, (max-width: 1365px) 30vw, 400px',
    QUIZ_2_IN_LINE: '(max-width: 768px) 40vw, (max-width: 1365px) 45vw, 600px',
  },
  MEAL_PLANNER: {
    INTRODUCTION: '(max-width: 1365px) 100vw, 60vw',
  },
  RECIPE_LISTINGS: {
    STANDARD: '(max-width: 768px) 50vw, (max-width: 1366px) 30vw, 400px',
    NON_STANDARD: '(max-width: 768px) 100vw, (max-width: 1366px) 50vw, 500px',
    MEAL_PLANNER: '(max-width: 1366px) 50vw, 400px',
  },
  PAGE_LISTINGS: {
    CAROUSEL: '(max-width: 768px) 30vw, (max-width: 1366px) 25vw, 300px',
    TILES: '(max-width: 1366px) 50vw, 500px',
  },
  HERO: '(max-width: 1365px) 500px, 800px',
  PROMISE_HELLMANNS:
    '(max-width: 767px) 100vw, (max-width: 1365px) 50vw, 500px',
  CONTAINER_WIDTH: '(max-width: 1365px) 100vw, 1092px',
  FULL_WIDTH: '100vw',
};
