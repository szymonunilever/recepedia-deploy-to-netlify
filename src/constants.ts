/**
 * RecipePersonalizationFormula
 * template - query string which using for Recipe Personalization result.
 * where:
 * Q - intro Quiz Answers
 * MP - meal Planer Answers
 * #(digit) - number of question
 * ^(digit) - weight of condition
 * searchAttributes - fields identifications
 * */
export const RecipePersonalizationFormula: RecipePersonalizationFormulaProps = {
  template: 'Q#2^4 AND (MP#1^3 OR MP#3^3 OR MP#2^2 OR Q#4^2 OR MP#4)',
  searchAttributes: {
    tags: 'tagGroups.tags.id',
    cookTime: 'recipeDetails.cookTime',
    serves: 'recipeDetails.serves',
  },
};

export interface RecipePersonalizationFormulaProps {
  template: string;
  searchAttributes: {
    [key: string]: string;
  };
}
