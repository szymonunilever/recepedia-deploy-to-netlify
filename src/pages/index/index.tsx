import { graphql } from 'gatsby';
import React from 'react';
import Layout from 'src/components/Layout/Layout';
import SEO from 'src/components/Seo/Seo';
import { Text, TagName } from 'src/components/lib/components/Text';

import { RecipeListing } from 'src/components/lib/components/RecipeListing';
import Hero from 'src/components/lib/components/Hero';
import { findPageComponentContent } from 'src/utils';
import { RecipeItem } from 'src/components/lib/components/RecipeListing/partials';

const HomePage = ({ data }: HomePageProps) => {
  const page = data.allPage.edges[0].node;
  const components = page.components.items;
  const recipes = data.allRecipe.nodes;

  return (
    <Layout>
      <SEO title="Recepedia Home" />

      <Text tag={TagName['h1']} text={data.allPage.edges[0].node.title} />
      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'LatestAndGreatest'
          )}
          list={recipes}
        />
      </section>

      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'TopRecipes'
          )}
          list={recipes}
        />
      </section>

      <section>
        <Hero
          content={findPageComponentContent(components, 'Hero')}
          viewType="Image"
        />
      </section>
    </Layout>
  );
};

export default HomePage;

export const pageQuery = graphql`
  {
    allPage(filter: { type: { eq: "Home" } }) {
      edges {
        node {
          ...PageFields
        }
      }
    }
    allRecipe(skip: 10) {
      nodes {
        ...RecipeFields
      }
    }
  }
`;

interface HomePageProps {
  data: {
    allPage: {
      edges: Edge<PageNode>[];
    };
    allRecipe: {
      nodes: RecipeItem[];
    };
  };
}

interface Edge<T> {
  node: T;
}

interface PageNode {
  components: {
    items: {
      name: string;
      content: any;
    }[];
  };
  title: string;
  type: string;
}
