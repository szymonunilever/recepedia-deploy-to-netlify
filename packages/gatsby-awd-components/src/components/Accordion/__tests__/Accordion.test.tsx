import { mount, ReactWrapper } from 'enzyme';
import React, { ReactNode } from 'react';
import toJson from 'enzyme-to-json';

import { KeyCode } from '../../constants';
import { Accordion, AccordionProps } from '../index';

describe('<Accordion />', () => {
  let wrapper: ReactWrapper;
  const children: ReactNode = React.createElement('div', null, 'Test');
  const props: AccordionProps = {
    title: {
      name: 'Name',
      label: 'Label',
    },
    isOpen: false,
    children: [],
    className: 'className',
    Icon: 'svg',
    IconOpened: 'svg',
  };

  beforeEach(() => {
    wrapper = mount(<Accordion {...props}>{children}</Accordion>);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Has to toggle menu onClick', () => {
    expect(wrapper.find('.accordion').hasClass('is-opened')).toBe(false);
    wrapper.find('.accordion__title').simulate('click');
    expect(wrapper.find('.accordion').hasClass('is-opened')).toBe(true);
  });

  it('Has to toggle menu onKeyUp with enter/space key', () => {
    const event = {
      key: 'Enter',
      keyCode: KeyCode.Enter,
      which: KeyCode.Enter,
    };

    expect(wrapper.find('.accordion').hasClass('is-opened')).toBe(false);
    wrapper.find('.accordion__title').simulate('keyUp', event);
    expect(wrapper.find('.accordion').hasClass('is-opened')).toBe(true);
  });

  it('Has no to toggle menu onKeyUp with no enter/space key', () => {
    const event = {
      preventDefault: () => {},
      key: 'Tab',
      keyCode: KeyCode.Tab,
      which: KeyCode.Tab,
    };

    expect(wrapper.find('.accordion').hasClass('is-opened')).toBe(false);
    wrapper.find('.accordion__title').simulate('keyUp', event);
    expect(wrapper.find('.accordion').hasClass('is-opened')).toBe(false);
  });
});
