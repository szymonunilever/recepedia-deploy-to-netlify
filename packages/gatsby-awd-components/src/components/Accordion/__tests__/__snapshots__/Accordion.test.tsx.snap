// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> matches the snapshot 1`] = `
<Component
  Icon="svg"
  IconOpened="svg"
  className="className"
  isOpen={false}
  title={
    Object {
      "label": "Label",
      "name": "Name",
    }
  }
>
  <div
    className="accordion className"
    data-componentname="accordion"
    data-componentview="accordion"
  >
    <div
      aria-expanded={false}
      aria-label="Label"
      className="accordion__title"
      id="Name"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="tree"
      tabIndex={0}
    >
      Label
      <svg />
    </div>
    <div
      className="accordion__body"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</Component>
`;
